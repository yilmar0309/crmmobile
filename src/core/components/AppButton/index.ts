export {AppButton, AppButtonVariant, AppButtonSizeVariant} from './AppButton';
