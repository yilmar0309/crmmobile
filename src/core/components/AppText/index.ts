export {AppText, AppTextVariant} from './AppText';
