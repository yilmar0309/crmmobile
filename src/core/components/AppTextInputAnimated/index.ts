export {AppTextInputAnimated, TextInputState} from './AppTextInputAnimated';
