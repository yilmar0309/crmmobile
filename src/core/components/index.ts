export {AppText, AppTextVariant} from './AppText';
export {AppButton, AppButtonVariant, AppButtonSizeVariant} from './AppButton';
export {AppTextInputAnimated, TextInputState} from './AppTextInputAnimated';
export {AppContainer} from './AppContainer';
export {AppSpacer} from './AppSpacer';
export {AppHeaderLeftBack} from './AppHeaderLeftBack';
export {AppShowEmptyList} from './AppShowEmptyList';
export {AppLoading} from './AppLoading';
export {AppImage} from './AppImage';
export {AppLoadingList} from './AppLoadingList';
