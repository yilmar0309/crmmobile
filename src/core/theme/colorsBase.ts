export const colorsBase = {
  WHITE: '#FFFFFF',
  BLACK: '#000000',
  GRAY: '#F8F8F8',
};
