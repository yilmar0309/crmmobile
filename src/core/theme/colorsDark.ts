export const colorsDark = {
  BACKGROUND: '#2C3E50',
  ON_BACKGROUND: '#E1E2E1',

  // Background primary color
  PRIMARY_COLOR: '#4d9ecf',
  ON_PRIMARY: '#FFFFFF',
  SECONDARY_COLOR: '#356ab0',

  // Background
  BACKGROUND_SCREEN_COLOR: '#2C3E50',

  // SURFACE
  SURFACE_1: '#34495E',

  // CONTENT
  CONTENT_PRIMARY: '#E1E2E1',
  CONTENT_SECONDARY: '#AAB0B6',

  // Text color
  PRIMARY_TEXT_COLOR: '#E1E2E1',
  SECONDARY_TEXT_COLOR: '#AAB0B6',
  DISABLED_TEXT_COLOR: '#52555A',
  TEXT_SIMPLE_COLOR: '#34495E',

  // TextInput color
  BACKGROUND_TEXT_INPUT_COLOR: '#34495E',

  // Button
  BACKGROUND_BUTTON_COLOR: '#2C3E50',
  BUTTON_TEXT_CONTAINED_COLOR: '#FFFFFF',
  BUTTON_TEXT_OUTLINED_COLOR: '#67b7ed',
  BUTTON_TEXT_COLOR: '#FFFFFF',
  BUTTON_BORDER_COLOR: '#67b7ed',
  BUTTON_DISABLED_COLOR: '#AAB0B6',
  BUTTON_DISABLED_TEXT_COLOR: '#CDD1D0',

  // Bottom Tab
  BOTTOM_TAB_COLOR: '#E1E2E1',

  // GRAY
  GRAY_01: '#34495E',
  GRAY_02: '#273746',
  GRAY_03: '#1B2C3B',
  GRAY_04: '#22313F',
  GRAY_05: '#293B4D',
  GRAY_06: '#999999',

  // Alert color
  SUCCESS_COLOR: '#2D7D46',
  WARNING_COLOR: '#C97B00',
  ERROR_COLOR: '#D0342C',
  INFO_COLOR: '#8D8B92',

  WHATSAPP: '#1E7D47',
  FACEBOOK: '#1D508D',
  INSTAGRAM: '#AD2671',
  TIKTOK: '#181818',

  // Basic color
  WHITE: '#FFFFFF',
  BLACK: '#121212',

  OUTLINE: '#2C313C',

  // SENTIMENT
  SENTIMENT_SUCCESS_BACKGROUND: '#1B4026',
  SENTIMENT_SUCCESS: '#2D7D46',
  SENTIMENT_INFO_BACKGROUND: '#2C3E50',
  SENTIMENT_INFO: '#1E2620',
  SENTIMENT_WARNING_BACKGROUND: '#332B1A',
  SENTIMENT_WARNING: '#C97B00',
  SENTIMENT_NEGATIVE_ERROR_BACKGROUND: '#401311',
  SENTIMENT_NEGATIVE_COLOR: '#D0342C',
  SENTIMENT_TEXT: '#1E2620',
  SENTIMENT_POSITIVE_BG: '#1B4026',
  SENTIMENT_WARNING_BACKGROUND_LOCK: '#664D1A',
  SENTIMENT_WARNING_ON_WARNING_BG: '#3A2800',

  // MAPS
  COLOR_MAP_LOSSES: '#FF6B6B',
  COLOR_MAP_FOUND: '#4ECDC4',
  COLOR_MAP_ADOPTION: '#FFD93D',
};
