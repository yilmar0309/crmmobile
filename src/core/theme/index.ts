export {colorsBase} from './colorsBase';
export {colorsLight} from './colorsLight';
export {colorsDark} from './colorsDark';
