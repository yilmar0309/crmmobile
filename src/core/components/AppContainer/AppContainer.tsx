import React from 'react';
import {
  KeyboardAvoidingView,
  Platform,
  SafeAreaView,
  StyleProp,
  StyleSheet,
  useColorScheme,
  ViewStyle,
} from 'react-native';
import {useHeaderHeight} from '@react-navigation/elements';
import {colorsDark, colorsLight} from '@/core/theme';

interface IAppContainer {
  children: React.ReactNode;
  style?: StyleProp<ViewStyle>;
  avoidKeyboard?: boolean;
  paddingVertical?: number;
}

export const AppContainer = ({
  children,
  style,
  avoidKeyboard = false,
  paddingVertical = 8,
}: IAppContainer) => {
  const headerHeight = useHeaderHeight();
  const isDarkMode = useColorScheme() === 'dark';

  return (
    <SafeAreaView
      style={[
        styles.container,
        {
          backgroundColor: isDarkMode
            ? colorsDark.BACKGROUND_SCREEN_COLOR
            : colorsLight.BACKGROUND_SCREEN_COLOR,
        },
        style,
      ]}
      testID="app-container">
      {avoidKeyboard ? (
        <KeyboardAvoidingView
          behavior={Platform.OS === 'ios' ? 'padding' : undefined}
          enabled
          style={[style, styles.container, {paddingVertical}]}
          keyboardVerticalOffset={headerHeight}
          testID="keyboard-avoiding-view">
          {children}
        </KeyboardAvoidingView>
      ) : (
        children
      )}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
