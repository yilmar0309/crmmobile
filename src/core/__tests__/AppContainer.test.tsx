import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {render} from '@testing-library/react-native';
import {AppContainer} from '../components/AppContainer';
import {colorsDark, colorsLight} from '../theme';
const mockedAppearance = require('react-native/Libraries/Utilities/Appearance');

describe('AppContainer', () => {
  beforeEach(() => {
    mockedAppearance.setColorScheme('dark');
  });

  afterEach(() => {
    mockedAppearance.setColorScheme('light');
  });

  it('renders correctly in dark mode', () => {
    const {getByText, toJSON} = render(
      <AppContainer>
        <Text>Test Content</Text>
      </AppContainer>,
    );

    expect(getByText('Test Content')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies dark mode background color', () => {
    const {getByTestId} = render(
      <AppContainer>
        <Text>Child Content</Text>
      </AppContainer>,
    );

    const parent = getByTestId('app-container');
    const flattenedStyles = StyleSheet.flatten(parent.props.style);
    expect(flattenedStyles.backgroundColor).toBe(
      colorsDark.BACKGROUND_SCREEN_COLOR,
    );
  });
  it('applies light mode background color', () => {
    mockedAppearance.setColorScheme('light');

    const {getByTestId} = render(
      <AppContainer>
        <Text>Child Content</Text>
      </AppContainer>,
    );

    const parent = getByTestId('app-container');
    const flattenedStyles = StyleSheet.flatten(parent.props.style);
    expect(flattenedStyles.backgroundColor).toBe(
      colorsLight.BACKGROUND_SCREEN_COLOR,
    );
  });

  it('does not include keyboard avoiding behavior on Android', () => {
    jest.mock('react-native/Libraries/Utilities/Platform', () => ({
      OS: 'android',
      select: jest.fn(),
    }));

    const {getByText} = render(
      <AppContainer avoidKeyboard={true}>
        <Text>Test Content</Text>
      </AppContainer>,
    );
    const parent = getByText('Test Content').parent;
    expect(parent?.props.behavior).toBeUndefined();
  });
});
