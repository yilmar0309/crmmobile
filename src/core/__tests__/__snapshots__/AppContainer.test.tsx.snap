// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer renders correctly in dark mode 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flex": 1,
      },
      {
        "backgroundColor": "#2C3E50",
      },
      undefined,
    ]
  }
  testID="app-container"
>
  <Text>
    Test Content
  </Text>
</RCTSafeAreaView>
`;
